src/encoded/static/components/view_controls.js

```diff
@@ -201,86 +94,38 @@ const getQueryFromFilters = (filters) => {
 /**
  * Displays view control buttons appropriate for the given search results.
  */
-export const ViewControls = ({ results, filterTerm, activeFilters, alternativeNames }) => {
-    // Add the hard-coded type filter if given.
-    let filters = [];
-    if (filterTerm) {
-        filters = results.filters.concat({ field: 'type', term: filterTerm });
-    } else {
-        filters = results.filters;
-    }
-
-    // if we have 'activeFilters' set on mouse development matrix page, append these to view control button links
-    if (activeFilters) {
-        activeFilters.forEach((f) => {
-            if (['adult', 'postnatal', 'embryo'].includes(f)) {
-                const stageTerm = f === 'embryo' ? 'embryonic' : f;
-                const stageFilterExists = filters.filter(f2 => f2.term === stageTerm).length > 0;
-                if (!stageFilterExists) {
-                    filters.push({
-                        term: f === 'embryo' ? 'embryonic' : f,
-                        remove: '',
-                        field: 'replicates.library.biosample.life_stage',
-                    });
-                }
-            } else {
-                const stageTerm = f.split(' ')[0] === 'embryo' ? 'embryonic' : f.split(' ')[0];
-                const ageTerm = f.split(' ').slice(1).join(' ');
-                const stageFilterExists = filters.filter(f2 => f2.term === stageTerm).length > 0;
-                const ageFilterExists = filters.filter(f2 => f2.term === ageTerm).length > 0;
-                if (!stageFilterExists) {
-                    filters.push({
-                        term: stageTerm,
-                        remove: '',
-                        field: 'replicates.library.biosample.life_stage',
-                    });
-                }
-                if (!ageFilterExists) {
-                    filters.push({
-                        term: ageTerm,
-                        remove: '',
-                        field: 'replicates.library.biosample.age_display',
-                    });
-                }
-            }
-        });
-    }
+export const ViewControls = ({ results, additionalFilters }) => {
```

Comment:
Since the parameters can come from outside the file, I do think you need to verify the variables are usable. For example, results can be null and additionalFilter can be a number.